src/sdJwt/sdJwt.ts

```diff
@@ -305,89 +307,65 @@ export class SdJwt<
-     *
-     * @throws when indices are supplied, but there are no disclosures on the instance
-     * @throws when the indices list is larger than the disclosure list
-     * @throws when a negative index is provided
-     * @throws when `NaN` or `Infinity` is supplied
-     * @throws when an index larger than the `disclosures.length - 1` is provided
-     * @throws when the same index is included multiple times
+     * @example
+     * The following example will expose `name`, `a.nested`, and `orderItems[0]` and `orderItems[2]`.
+     * Based on the disclosures it will also expose the parent and child disclosures when needed.
+     * E.g. if `a` can only be disclosed as a whole, disclosing `a.nested` will also disclose `a`.
+     * The same is true for child disclosures. If you expose `name`, and it potentially contains recursive
+     * disclosures, all disclosures under name will be disclosed as well.
+     * ```ts
+     * await sdJwt.present({
+     *   name: true,
+     *   a: {
+     *     nested: 'property'
+     *   }
+     *   orderItems: [true, false, true]
+     * })
+     * ```
+     *
+     * @throws when the presentation frame does not match the decoded/pretty payload of the sd-jwt
+     * @throws when the presentation frame contains fields other than object, array or boolean
      *
      */
-    public async present(includedDisclosureIndices?: Array<number>) {
+    public async present(presentationFrame?: PresentationFrame<Payload>) {
         if (!this.disclosures && this.disclosureFrame) {
             await this.applyDisclosureFrame()
         }
 
-        if (
-            includedDisclosureIndices &&
-            includedDisclosureIndices.length > 0 &&
-            !this.disclosures
-        ) {
-            throw new SdJwtError(
-                'Cannot create a presentation with disclosures while no disclosures are on the sd-jwt'
-            )
-        }
-
-        if (
-            includedDisclosureIndices &&
-            includedDisclosureIndices.length > this.disclosures!.length
-        ) {
-            throw new SdJwtError(
-                `List of included indices (${
-                    includedDisclosureIndices.length
-                }) is longer than the list of disclosures (${
-                    this.disclosures!.length
-                }).`
-            )
-        }
-
-        const maxIndex = Math.max(...(includedDisclosureIndices ?? []))
-        const minIndex = Math.min(...(includedDisclosureIndices ?? []))
-
-        if (minIndex < 0) {
-            throw new SdJwtError('Only positive indices are allowed')
-        }
-
-        if (
-            (includedDisclosureIndices ?? []).some(
-                (i) => isNaN(i) || !isFinite(i)
-            )
-        ) {
-            throw new SdJwtError('NaN and infinity are not allowed as indices')
-        }
+        // TODO: wouldn't it be easier if this returned the value so we don't have to use !
```

Comment:
It could return `this` with the property defined or the value. Any preference?

Reply:
I think the value would be great.

Then it becomes:

```
const payload = this.assertPayload()
```

Reply:
Yeah I think that is something I can add in a follow-up PR. Bothered me quite a bit as well.